android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt

```diff
@@ -27,35 +25,53 @@ class ListingDetailsPageViewModel @Inject constructor(
 		BehaviorSubject.createDefault(checkNotNull(savedStateHandle["listingId"]))
 	private val imageIdsStream: BehaviorSubject<List<String>> = BehaviorSubject.createDefault(emptyList())
 	val favouritedStream: BehaviorSubject<Boolean> = BehaviorSubject.createDefault(false)
-	private val listingDetailsStream: Observable<ListingDetails> = listingIdStream.map {
-		runBlocking {
-			val listing = defaultApi.listingsDetails(it)
-			favouritedStream.onNext(listing.favourited)
-			imageIdsStream.onNext(listing.details.imgIds)
-			listing.details
+	private val listingDetailsStream: Observable<Result<ListingDetails>> = listingIdStream.map {
+		runCatching {
+			runBlocking {
+				val listing = defaultApi.listingsDetails(it)
+				favouritedStream.onNext(listing.favourited)
+				imageIdsStream.onNext(listing.details.imgIds)
+				listing.details
+			}
 		}
 	}
+		.onErrorResumeWith(Observable.never())
 		.subscribeOn(Schedulers.io())
+
 	private val imagesStream: Observable<List<Bitmap>> = imageIdsStream.map {
-		runBlocking {
-			it.map { id ->
-				val encodedImage = defaultApi.listingsImagesGet(id)
-				encodedImage.base64ToBitmap()
+		runCatching {
+			runBlocking {
+				it.map { id ->
+					defaultApi.listingsImagesGet(id)
+				}
 			}
 		}
 	}
-		.subscribeOn(Schedulers.computation())
+		.filter { it.isSuccess }
+		.map {
+			it.getOrDefault(emptyList())
+		}
+		.subscribeOn(Schedulers.io())
+		.observeOn(Schedulers.computation())
+		.map { base64ImageList ->
+			base64ImageList.map { it.base64ToBitmap() }
+		}
+		.observeOn(Schedulers.io())
```

Comment:
Basically this mess of `Schedulers` ensures that the api calls are done on the IO threads and the `base64ToBitmap` are done on the computation threads